velox/common/compression/CMakeLists.txt

```diff
@@ -16,8 +16,17 @@ if(${VELOX_BUILD_TESTING})
   add_subdirectory(tests)
 endif()
 
-velox_add_library(velox_common_compression Compression.cpp LzoDecompressor.cpp)
+set(VELOX_COMMON_COMPRESSION_SRCS Compression.cpp LzoDecompressor.cpp)
+set(VELOX_COMMON_COMPRESSION_LINK_LIBS velox_status Folly::folly)
+
+if(VELOX_ENABLE_COMPRESSION_LZ4)
+  list(APPEND VELOX_COMMON_COMPRESSION_SRCS Lz4Compression.cpp
+       HadoopCompressionFormat.cpp)
+  list(APPEND VELOX_COMMON_COMPRESSION_LINK_LIBS lz4::lz4)
+endif()
+
```

Comment:
Please avoid using variables in this way. Instead  use `velox_target_sources`  and `velox_link_libraries` instead. Both can be called multiple times on the same target.